pkg/virt-operator/resource/generate/install/strategy.go

```diff
@@ -292,42 +368,28 @@ func GenerateCurrentInstallStrategy(config *operatorutil.KubeVirtDeploymentConfi
 		}
 	}
 
-	var productName string
-	var productVersion string
-
-	if isValidLabel(config.GetProductName()) {
-		productName = config.GetProductName()
-	} else {
-		log.Log.Errorf("invalid kubevirt.spec.productName: labels must be 63 characters or less, begin and end with alphanumeric characters, and contain only dot, hyphen or dash")
-	}
-	if isValidLabel(config.GetProductVersion()) {
-		productVersion = config.GetProductVersion()
-	} else {
-		log.Log.Errorf("invalid kubevirt.spec.productVersion: labels must be 63 characters or less, begin and end with alphanumeric characters, and contain only dot, hyphen or dash")
-	}
-
```

Comment:
there's a lot of indirection going on here, but ultimately this corresponds to the kv.Spec.ProductVersion field on the kubevirt cr. I think we still need this. Maybe this was simply moved and I'm missing it?

Reply:
From my understanding this code was redundant, . We are [injecting these values on reconcile](https://github.com/kubevirt/kubevirt/blob/ff2218cb391acafae35a094c0303ebc0536da490/pkg/virt-operator/resource/apply/reconcile.go#L87-L110). Having them written in the configmap seems confusing. If the operator removes them from the KV config they will then persist, but only for deployments and daemonsets. However, on second inspection it looks like product version is always filled for these resources with the KubeVirt version if the product version is not set. So I will add this back. But it does seem a bit confusing and disjointed.

Reply:
Ohhhh, i get it. yep I agree.